lectures/textures/SDLProject/main.cpp

```diff
@@ -0,0 +1,288 @@
+    
+    g_display_window = SDL_CreateWindow("Hello, Textures!",
+                                      SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
+                                      WINDOW_WIDTH, WINDOW_HEIGHT,
+                                      SDL_WINDOW_OPENGL);
+    
+    SDL_GLContext context = SDL_GL_CreateContext(g_display_window);
+    SDL_GL_MakeCurrent(g_display_window, context);
+    
+#ifdef _WINDOWS
+    glewInit();
+#endif
+    
+    glViewport(VIEWPORT_X, VIEWPORT_Y, VIEWPORT_WIDTH, VIEWPORT_HEIGHT);
+    
+    g_shader_program.load(V_SHADER_PATH, F_SHADER_PATH);
+    
+    g_model_matrix = glm::mat4(1.0f);
+    g_view_matrix = glm::mat4(1.0f);  // Defines the position (location and orientation) of the camera
+    g_projection_matrix = glm::ortho(-5.0f, 5.0f, -3.75f, 3.75f, -1.0f, 1.0f);  // Defines the characteristics of your camera, such as clip planes, field of view, projection method etc.
+    
+    g_shader_program.set_projection_matrix(g_projection_matrix);
+    g_shader_program.set_view_matrix(g_view_matrix);
+    // Notice we haven't set our model matrix yet!
+    
+    glUseProgram(g_shader_program.get_program_id());
+    
+    glClearColor(BG_RED, BG_BLUE, BG_GREEN, BG_OPACITY);
+    
+    g_player_texture_id = load_texture(PLAYER_SPRITE_FILEPATH);
+    
+    // enable blending
+    glEnable(GL_BLEND);
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+}
+
+void process_input()
+{
+    g_player_movement = glm::vec3(0.0f);
+    
+    SDL_Event event;
+    
+    while (SDL_PollEvent(&event))
+    {
+        switch (event.type) {
+            case SDL_WINDOWEVENT_CLOSE:
+            case SDL_QUIT:
+                g_app_status = TERMINATED;
+                break;
+                
+            case SDL_KEYDOWN:
+                switch (event.key.keysym.sym) {
+                    case SDLK_RIGHT:
+                        g_player_movement.x = 1.0f;
+                        break;
+                    case SDLK_LEFT:
+                        g_player_movement.x = -1.0f;
+                        break;
+                    case SDLK_q:
+                        g_app_status = TERMINATED;
+                        break;
+                    default:
+                        break;
+                }
+            default:
+                break;
+        }
+    }
+    
+    const Uint8 *key_states = SDL_GetKeyboardState(NULL); // array of key states [0, 0, 1, 0, 0, ...]
+    
+    if (key_states[SDL_SCANCODE_LEFT])
+    {
+        g_player_movement.x = -1.0f;
+    } else if (key_states[SDL_SCANCODE_RIGHT])
+    {
+        g_player_movement.x = 1.0f;
+    }
+    
+    if (key_states[SDL_SCANCODE_UP])
+    {
+        g_player_movement.y = 1.0f;
+    } else if (key_states[SDL_SCANCODE_DOWN])
+    {
+        g_player_movement.y = -1.0f;
+    }
+    
+    if (glm::length(g_player_movement) > 1.0f)
+    {
+        g_player_movement = glm::normalize(g_player_movement);
+    }
```

Comment:
Again, this is from next week's lectures. Replace with Tuesday's `process_input`.